File: Lista II - Aula 4/J1_08/Empresa.java
public class Empresa {
    String nome;
    String cnpj;
    Departamento[] departamentos;

    Empresa() {}

    Empresa (int num){
        this.departamentos = new Departamento[num];
    }

    public void setNome (String nome){
        this.nome = nome;
    }

    String getNome(){
        return this.nome;
    }

    public void setCNPJ (String cnpj){
        this.cnpj = cnpj;
    }

    String getCNPJ(){
        return this.cnpj;
    }

    public void setDepartamento (int i, Departamento departamento){
        this.departamentos[i] = departamento;
    }

    Departamento getDepartamento(int i){
        return this.departamentos[i];
    }
}

File: Lista V (C++) - Aula 22/C1_02/Empresa.cpp
#include <iostream>
#include "Empresa.h"
#include "NumberUtils.h"

//Construtores
Empresa::Empresa(const string& nome, const string& cnpj) {
	this->nome = nome;
	this->cnpj = cnpj;
}

//Métodos
void Empresa::adicionarDepartamento(Departamento* departamento) {
	departamentos.push_back(departamento);
}

void Empresa::darAumento(const int idxDepto, const double percentual) {
	departamentos[idxDepto]->darAumento(percentual);
}

void Empresa::transferirFuncionario(const int idxDeptoOrigem, const int idxFunc, const int idxDeptoDestino) {
	Funcionario* f = departamentos[idxDeptoOrigem]->getFuncionario(idxFunc);
	departamentos[idxDeptoDestino]->adicionarFuncionario(f);
	departamentos[idxDeptoOrigem]->removerFuncionario(idxFunc);
}

Empresa::~Empresa() {
	for (auto d : departamentos) //for para todo departamento d do vetor
		delete d;
}

ostream& operator<<(ostream &out, const Empresa &empresa) {
	for (auto& d : empresa.departamentos)
		out << d->getNome() << " R$ " << formatDoubleCurrency(d->calcularCusto(), LOCALE_PT_BR) << endl;
	return out;
}

